lib/Db/RowCellSelectionMapper.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace OCA\Tables\Db;
+
+use OCP\IDBConnection;
+
+/**
+ * @template-extends RowCellMapperSuper<RowCellSelection, string, string|array>
+ */
+class RowCellSelectionMapper extends RowCellMapperSuper {
+	protected string $table = 'tables_row_cells_selection';
+
+	public function __construct(IDBConnection $db) {
+		parent::__construct($db, $this->table, RowCellSelection::class);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function parseValueIncoming(Column $column, $value): string {
+		if ($column->getSubtype() === 'check') {
+			return json_encode(ltrim($value, '"'));
+		}
+
+		if ($column->getSubtype() === '' || $column->getSubtype() === null) {
+			return $value;
```

Comment:
This causes a new issue on the miration:

> TypeError: OCA\Tables\Db\RowCellSelectionMapper::parseValueIncoming(): Return value must be of type string, null returned in /var/www/html/apps-shared/tables/lib/Db/RowCellSelectionMapper.php:26

```
Stack trace:
#0 /var/www/html/apps-shared/tables/lib/Db/Row2Mapper.php(685): OCA\Tables\Db\RowCellSelectionMapper->parseValueIncoming(Object(OCA\Tables\Db\Column), NULL)
#1 /var/www/html/apps-shared/tables/lib/Db/Row2Mapper.php(609): OCA\Tables\Db\Row2Mapper->formatValue(Object(OCA\Tables\Db\Column), NULL, 'in')
#2 /var/www/html/apps-shared/tables/lib/Db/Row2Mapper.php(496): OCA\Tables\Db\Row2Mapper->insertCell(669, 571, NULL, '2023-06-21 09:5...', 'philipp')
#3 /var/www/html/apps-shared/tables/lib/Db/LegacyRowMapper.php(448): OCA\Tables\Db\Row2Mapper->insert(Object(OCA\Tables\Db\Row2), Array)
#4 /var/www/html/apps-shared/tables/lib/Migration/NewDbStructureRepairStep.php(97): OCA\Tables\Db\LegacyRowMapper->transferLegacyRow(Object(OCA\Tables\Db\LegacyRow), Array)
#5 /var/www/html/apps-shared/tables/lib/Migration/NewDbStructureRepairStep.php(74): OCA\Tables\Migration\NewDbStructureRepairStep->transferTable(Object(OCA\Tables\Db\Table), Object(OC\Repair))
#6 /var/www/html/apps-shared/tables/lib/Migration/NewDbStructureRepairStep.php(61): OCA\Tables\Migration\NewDbStructureRepairStep->transferDataForTables(Array, Object(OC\Repair))
#7 /var/www/html/lib/private/Repair.php(127): OCA\Tables\Migration\NewDbStructureRepairStep->run(Object(OC\Repair))
#8 /var/www/html/lib/private/legacy/OC_App.php(838): OC\Repair->run()
#9 /var/www/html/lib/private/legacy/OC_App.php(779): OC_App::executeRepairSteps('tables', Array)
#10 /var/www/html/lib/private/Updater.php(360): OC_App::updateApp('tables')
#11 /var/www/html/lib/private/Updater.php(272): OC\Updater->doAppUpgrade()
#12 /var/www/html/lib/private/Updater.php(140): OC\Updater->doUpgrade('29.0.0.4', '29.0.0.4')
#13 /var/www/html/core/Command/Upgrade.php(213): OC\Updater->upgrade()
#14 /var/www/html/apps-extra/collectives/vendor/symfony/console/Command/Command.php(298): OC\Core\Command\Upgrade->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /var/www/html/apps-extra/collectives/vendor/symfony/console/Application.php(1040): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /var/www/html/apps-extra/collectives/vendor/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand(Object(OC\Core\Command\Upgrade), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/html/apps-extra/collectives/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/html/lib/private/Console/Application.php(206): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/html/console.php(100): OC\Console\Application->run()
#20 /var/www/html/occ(11): require_once('/var/www/html/c...')
#21 {main}%
```

Reply:
Also I think we should still catch any errors in #5 /var/www/html/apps-shared/tables/lib/Migration/NewDbStructureRepairStep.php(74): OCA\Tables\Migration\NewDbStructureRepairStep->transferTable(Object(OCA\Tables\Db\Table), Object(OC\Repair)) and continue